webpack/ForemanTasks/Components/TaskDetails/TaskDetailsActions.js

```diff
@@ -1,143 +1,75 @@
+import { post, get } from 'foremanReact/redux/API';
 import { addToast } from 'foremanReact/redux/actions/toasts';
 import { translate as __ } from 'foremanReact/common/I18n';
 import {
-  FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST,
-  FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS,
-  FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE,
-  FOREMAN_TASK_DETAILS_STOP_POLLING,
-  FOREMAN_TASK_DETAILS_START_POLLING,
-  TASK_STEP_CANCEL_REQUEST,
-  TASK_STEP_CANCEL_FAILURE,
-  TASK_STEP_CANCEL_SUCCESS,
-} from './TaskDetailsConstants';
+  withInterval,
+  stopInterval,
+} from 'foremanReact/redux/middlewares/IntervalMiddleware';
+import { TASK_STEP_CANCEL, FOREMAN_TASK_DETAILS } from './TaskDetailsConstants';
 import {
   errorToastData,
   infoToastData,
   successToastData,
 } from '../common/ToastsHelpers';
 
-export const taskReloadStop = timeoutId => {
-  if (timeoutId) {
-    clearTimeout(timeoutId);
-  }
-  return {
-    type: FOREMAN_TASK_DETAILS_STOP_POLLING,
-    payload: { timeoutId },
-  };
-};
+export const taskReloadStop = () => stopInterval(FOREMAN_TASK_DETAILS);
 
-export const taskReloadStart = (
-  timeoutId,
-  refetchTaskDetails,
-  id,
-  loading = false
-) => {
-  if (!timeoutId) {
-    timeoutId = setInterval(
-      () => refetchTaskDetails(id, loading, timeoutId),
+export const taskReloadStart = id => dispatch => {
+  dispatch(
+    withInterval(
+      get({
+        key: FOREMAN_TASK_DETAILS,
+        url: `/foreman_tasks/api/tasks/${id}/details?include_permissions`,
+        handleSuccess: ({ data }) => {
+          if (data.state === 'stopped') {
+            dispatch(taskReloadStop(id));
+          }
+        },
+        handleError: () => {
+          dispatch(taskReloadStop());
+        },
+      }),
       5000
-    );
-  }
-  return {
-    type: FOREMAN_TASK_DETAILS_START_POLLING,
-    payload: { timeoutId },
-  };
-};
-
-export const refetchTaskDetails = (id, loading, timeoutId) => dispatch => {
-  if (!loading) {
-    showLoading();
-    dispatch(startRequest());
-    reloadTasksDetails(id, timeoutId, dispatch);
-  }
-};
-
-const reloadTasksDetails = async (id, timeoutId, dispatch) => {
-  try {
-    const { data } = await API.get(
-      `/foreman_tasks/api/tasks/${id}/details?include_permissions`
-    );
-    dispatch(requestSuccess(data));
-  } catch (error) {
-    dispatch(taskReloadStop(timeoutId));
-    dispatch(requestFailure(error));
-  } finally {
-    hideLoading();
-  }
+    )
+  );
 };
 
-export const fetchTaskDetails = (
-  id,
-  timeoutId,
-  refetchTaskDetailsAction
-) => async dispatch => {
-  showLoading();
-  dispatch(startRequest());
-  await getTasksDetails(id, dispatch, timeoutId, refetchTaskDetailsAction);
+export const fetchTaskDetails = id => dispatch => {
+  dispatch(
+    get({
+      key: FOREMAN_TASK_DETAILS,
+      url: `/foreman_tasks/api/tasks/${id}/details?include_permissions`,
+      handleSuccess: ({ data }) => {
+        if (data.state !== 'stopped') {
+          dispatch(taskReloadStart(id));
```

Comment:
just wondering, why do you want to start the interval when `data.state` is `stopped`?

Reply:
Because if it's stopped the task won't have new info as it's "finished". for example if it's running it can change state to stopped, but stopped won't change it's state

Reply:
> Because if it's stopped the task won't have new info as it's "finished"

interesting, so if it's "stopped" and "finished", any chance that something will get update while we do the polling?

Reply:
Nope, thats why we start the interval only when it's not stopped

Reply:
sorry, I meant to comment on https://github.com/theforeman/foreman-tasks/pull/564/files#diff-6c17b677553fd00bde1acad329076f55R24

though what if there is already a polling process running ? we don't want to trigger two polling processes on the same page for the same api path.